src/cmdlib.sh

```diff
@@ -553,16 +553,20 @@ runcompose_tree() {
     else
         local tarball="${workdir}/tmp/repo/commit.tar"
         rm -f "${tarball}"
-        runvm_with_cache -- /usr/lib/coreos-assembler/compose.sh \
-            "${tarball}" "${composejson}" "$@"
-        if [ ! -f "${tarball}" ]; then
+        runvm_with_cache \
+            -chardev "file,id=tarout,path=${tarball}" \
+            -device "virtserialport,chardev=tarout,name=tarout" -- \
+            /usr/lib/coreos-assembler/compose.sh \
+                "/dev/virtio-ports/tarout" "$@"
```

Comment:
do we no longer need to pass `"${composejson}"` here?

Reply:
Ahh. I see. We changed the semantics in `compose.sh`.

Reply:
Yeah, it's part of the tarball now so that `compose.sh` doesn't directly write to it over 9p.